timelock-impl/src/test/java/com/palantir/atlasdb/timelock/ConjureTimelockResourceTest.java

```diff
@@ -0,0 +1,170 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.when;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.Duration;
+import java.util.concurrent.ExecutionException;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.net.HostAndPort;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.palantir.atlasdb.http.RedirectRetryTargeter;
+import com.palantir.atlasdb.timelock.api.ConjureTimelockService;
+import com.palantir.conjure.java.api.errors.QosException;
+import com.palantir.leader.NotCurrentLeaderException;
+import com.palantir.lock.impl.TooManyRequestsException;
+import com.palantir.lock.remoting.BlockingTimeoutException;
+import com.palantir.lock.v2.LeaderTime;
+import com.palantir.tokens.auth.AuthHeader;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ConjureTimelockResourceTest {
+    private static final AuthHeader AUTH_HEADER = AuthHeader.valueOf("Bearer test");
+    private static final int REMOTE_PORT = 4321;
+    private static final URL LOCAL = url("https://localhost:1234");
+    private static final URL REMOTE = url("https://localhost:" + REMOTE_PORT);
+    private static final RedirectRetryTargeter TARGETER = RedirectRetryTargeter.create(
+            LOCAL,
+            ImmutableList.of(LOCAL, REMOTE));
+
+    private static final String NAMESPACE = "test";
+
+    @Mock private AsyncTimelockService timelockService;
+    @Mock private LeaderTime leaderTime;
+
+    private ConjureTimelockResource resource;
+    private ConjureTimelockService service;
+
+    @Before
+    public void before() {
+        resource = new ConjureTimelockResource(TARGETER, unused -> timelockService);
+        service = ConjureTimelockResource.jersey(TARGETER, unused -> timelockService);
+        when(timelockService.leaderTime()).thenReturn(leaderTime);
+    }
+
+    @Test
+    public void canGetLeaderTime() {
+        assertThat(Futures.getUnchecked(resource.leaderTime(AUTH_HEADER, NAMESPACE))).isEqualTo(leaderTime);
+    }
+
+    @Test
+    public void jerseyPropagatesExceptions() {
+        when(resource.leaderTime(AUTH_HEADER, NAMESPACE)).thenThrow(new BlockingTimeoutException(""));
```

Comment:
`resource.leaderTime` will never throw `BlockingTimeoutException` right? It will always give you back a failed `ListenableFuture` if I've read correctly

Reply:
unless there's a generated form that's calling `.get` on it, then fine, but still feels a bit weird

Reply:
replaced with the failed listenable future